OutOfSchool/OutOfSchool.EmailSender/EmailSenderJob.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Quartz;
+using SendGrid;
+using SendGrid.Helpers.Mail;
+
+namespace OutOfSchool.EmailSender;
+
+[DisallowConcurrentExecution]
+public class EmailSenderJob : IJob
+{
+    private readonly IOptions<EmailOptions> emailOptions;
+    private readonly ILogger<EmailSenderJob> logger;
+    private readonly ISendGridClient sendGridClient;
+
+    public EmailSenderJob(
+        IOptions<EmailOptions> emailOptions,
+        ILogger<EmailSenderJob> logger,
+        ISendGridClient sendGridClient)
+    {
+        this.emailOptions = emailOptions;
+        this.logger = logger;
+        this.sendGridClient = sendGridClient;
+    }
+
+    public async Task Execute(IJobExecutionContext context)
```

Comment:
(Exception handling) From [Quartz.Net Job documantation](https://www.quartz-scheduler.net/documentation/quartz-3.x/tutorial/more-about-jobs.html#jobexecutionexception) it's seen that 
- whole job has to be wrapped in try/catch
- you have to manually throw `JobExecutionException` if something goes wrong (i.e. when `response.StatusCode == HttpStatusCode.TooManyRequests`)